frontend/src/main/scala/bloop/bsp/BspServer.scala

```diff
@@ -0,0 +1,79 @@
+package bloop.bsp
+
+import java.net.ServerSocket
+import java.util.Locale
+import java.nio.file.{Files, Path}
+
+import bloop.cli.{BspProtocol, Commands}
+import bloop.engine.{ExecutionContext, State}
+import com.martiansoftware.nailgun.{NGUnixDomainServerSocket, NGWin32NamedPipeServerSocket}
+import monix.execution.{Cancelable, Scheduler}
+
+object BspServer {
+  private[bloop] val isWindows: Boolean =
+    System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows")
+
+  import java.net.InetSocketAddress
+  private sealed trait ConnectionHandle { def socket: ServerSocket }
+  private case class WindowsLocal(pipeName: String, socket: ServerSocket) extends ConnectionHandle
+  private case class UnixLocal(path: Path, socket: ServerSocket) extends ConnectionHandle
+  private case class Tcp(address: InetSocketAddress, socket: ServerSocket) extends ConnectionHandle
+
+  import Commands.ValidatedBsp
+  import monix.{eval => me}
+  private def initServer(cmd: ValidatedBsp, state: State): me.Task[ConnectionHandle] = me.Task {
+    cmd match {
+      case Commands.WindowsLocalBsp(pipeName, _) =>
+        state.logger.debug(s"Establishing server connection at pipe $pipeName")
+        WindowsLocal(pipeName, new NGWin32NamedPipeServerSocket(pipeName))
+      case Commands.UnixLocalBsp(socketFile, _) =>
+        state.logger.debug(s"Establishing server connection at $socketFile")
+        UnixLocal(socketFile, new NGUnixDomainServerSocket(socketFile.toString))
+      case Commands.TcpBsp(address, portNumber, name) =>
+        val socketAddress = new InetSocketAddress(address, portNumber)
+        state.logger.debug(s"Establishing server connection via TCP at ${socketAddress}")
+        // Use 0 instead of a concrete number to have an infinite timeout
+        Tcp(socketAddress, new java.net.ServerSocket(portNumber, 0, address))
```

Comment:
This starts the server rather than establish a connection, right? I think the logs should read something along the lines of `Starting server listening on ...`, WDYT?